tests/test_operator.py

```diff
@@ -213,6 +213,33 @@ def test_multiple_steppers(self, expr, exp_uindices, exp_mods):
         assert(i.indices[i.function._time_position].modulo == exp_mods[i.function.name]
                for i in flatten(retrieve_indexed(i) for i in exprs))
 
+    def test_lower_stepping_dims_with_mutiple_iterations(self):
+        """
+        Test lowering SteppingDimensions for a time dimension with
+        more than one iteration loop with different ModuloDimensions.
+        MFE for issue #1486
+        """
+        grid = Grid(shape=(4, 4))
+
+        f = Function(name="f", grid=grid, space_order=4)
+        g = Function(name="g", grid=grid, space_order=4)
+        h = TimeFunction(name="h", grid=grid, space_order=4, time_order=2)
+
+        eqn = [Eq(f, h + 1), Eq(g, f),
+               Eq(h.forward, h + g + 1)]
+
+        op = Operator(eqn)
+
+        for iter in [i for i in FindNodes(Iteration).visit(op) if i.dim.is_Time]:
```

Comment:
Maybe you could also assert against the number of expected Iterations? w/o the patch, we would obtain more Iterations right? that'd be a simple yet effective check

Reply:
The patch doesn't change the number of iterations. It just fix what is going to happen if there is more than one. The MFE would produce two iterations for time dimension with or without the patch. 

We could assert 2 time iterations, but I think this is more related to cluster fusion in `_specialize_clusters` (e.g. devito /core/cpu.py) which I guess could vary for some implementation. Let me know if you still think it's worth it.

Reply:
Gotcha.

Any chance we can then assert the expected computed value? since this would be the only test of this kind, it could be quite useful

Reply:
Done. Very simple assertion. Re-check please!

Reply:
perfect, thanks